nucleus/payara-modules/notification-core/src/main/java/fish/payara/nucleus/notification/domain/execoptions/NotifierConfigurationExecutionOptionsFactory.java

```diff
@@ -28,11 +29,19 @@ and Distribution License("CDDL") (collectively, the "License").  You
 @Singleton
 public class NotifierConfigurationExecutionOptionsFactory {
 
-    public NotifierConfigurationExecutionOptions build(NotifierConfiguration notifierConfiguration) {
+    public <T extends NotifierConfigurationExecutionOptions> T build(NotifierConfiguration notifierConfiguration) {
         if (notifierConfiguration instanceof LogNotifierConfiguration) {
             LogNotifierConfigurationExecutionOptions executionOptions = new LogNotifierConfigurationExecutionOptions();
             executionOptions.setEnabled(Boolean.parseBoolean(notifierConfiguration.getEnabled()));
-            return executionOptions;
+            return (T) executionOptions;
+        }
+        if (notifierConfiguration instanceof HipchatNotifierConfiguration) {
```

Comment:
I hate this instanceof stuff it messes with my OO trained brain and mashes up the dependencies. Can't we use some sort of factory pattern or builder pattern here?